provider/infoblox.go

```diff
@@ -0,0 +1,389 @@
+	return nil
+}
+
+func (p *InfobloxProvider) zones() ([]ibclient.ZoneAuth, error) {
+	var res, result []ibclient.ZoneAuth
+	obj := ibclient.NewZoneAuth(ibclient.ZoneAuth{})
+	err := p.client.GetObject(obj, "", &res)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, zone := range res {
+		if p.domainFilter.Match(zone.Fqdn) {
+			result = append(result, zone)
+		}
+	}
+
+	return result, nil
+}
+
+type infobloxChangeMap map[string][]*endpoint.Endpoint
+
+func (p *InfobloxProvider) mapChanges(zones []ibclient.ZoneAuth, changes *plan.Changes) (infobloxChangeMap, infobloxChangeMap, infobloxChangeMap) {
+	ignored := map[string]bool{}
+	created := infobloxChangeMap{}
+	deleted := infobloxChangeMap{}
+	updated := infobloxChangeMap{}
+
+	mapChange := func(changeMap infobloxChangeMap, change *endpoint.Endpoint) {
+		zone := p.findZone(zones, change.DNSName)
+		if zone == nil {
+			if _, ok := ignored[change.DNSName]; !ok {
+				ignored[change.DNSName] = true
+				logrus.Infof("Ignoring changes to '%s' because a suitable Infoblox DNS zone was not found.", change.DNSName)
+			}
+			return
+		}
+		// Ensure the record type is suitable
+		changeMap[zone.Fqdn] = append(changeMap[zone.Fqdn], change)
+	}
+
+	for _, change := range changes.Create {
+		mapChange(created, change)
+	}
+	for _, change := range changes.Delete {
+		mapChange(deleted, change)
+	}
+	for _, change := range changes.UpdateNew {
+		mapChange(updated, change)
+	}
+	for _, change := range changes.UpdateOld {
+		mapChange(deleted, change)
+	}
+
+	return created, deleted, updated
+}
+
+func (p *InfobloxProvider) findZone(zones []ibclient.ZoneAuth, name string) *ibclient.ZoneAuth {
+	var result *ibclient.ZoneAuth
+
+	// Go through every zone looking for the longest name (i.e. most specific) as a matching suffix
+	for idx := range zones {
+		zone := &zones[idx]
+		if strings.HasSuffix(name, zone.Fqdn) {
+			if result == nil || len(zone.Fqdn) > len(result.Fqdn) {
+				result = zone
+			}
+		}
+	}
+	return result
+}
+
+func (p *InfobloxProvider) newObject(e *endpoint.Endpoint, getRef bool) (obj ibclient.IBObject, ref string, err error) {
+	switch e.RecordType {
+	case endpoint.RecordTypeA:
+		var records []ibclient.RecordA
+		obj = ibclient.NewRecordA(
+			ibclient.RecordA{
+				Name:     e.DNSName,
+				Ipv4Addr: e.Target,
+			},
+		)
+		if getRef {
+			p.client.GetObject(obj, "", &records)
+			if len(records) == 1 {
+				ref = records[0].Ref
+			}
+		}
+		return obj, ref, nil
+	case endpoint.RecordTypeCNAME:
+		var records []ibclient.RecordCNAME
+		obj = ibclient.NewRecordCNAME(
+			ibclient.RecordCNAME{
+				Name:      e.DNSName,
+				Canonical: e.Target,
+			},
+		)
+		if getRef {
+			p.client.GetObject(obj, "", &records)
+			if len(records) == 1 {
+				ref = records[0].Ref
+			}
+		}
+		return obj, ref, nil
+	case endpoint.RecordTypeTXT:
+		var records []ibclient.RecordTXT
+		if target, err := strconv.Unquote(e.Target); err == nil && !strings.Contains(e.Target, " ") {
```

Comment:
why error is ignored in this case and why there is a need to check for empty space in `e.Target` ?

Reply:
See https://github.com/kubernetes-incubator/external-dns/pull/349#discussion_r142693639. I'll add a comment here as well.